### PR TITLE
WT-7679 Create an evergreen test for many-dhandle-stress

### PR DESCRIPTION
**Pull request by etienneptl**
Creation of a new evergreen buildvariant to run the many dhandles perf test.
The executed script will be merged in #6659.
The thresholds have been set high for now while we are waiting to gather more results.

### CODE REVIEW

File: test/evergreen.yml [hunk 1]
```diff
@@ -2386,6 +2386,24 @@ tasks:
...
+            PATH=/opt/mongodbtoolchain/v3/bin:$PATH ADD_CFLAGS="-ggdb -fPIC"
+      - command: shell.exec
+        params:
+          working_dir: "wiredtiger/bench/workgen/runner"
+          script: |
+            set -o errexit
+            set -o verbose
+            export "PATH=/opt/mongodbtoolchain/v3/bin:$PATH"
+            ./workgen_perf_check.sh many-dhandle-stress.py output.log 50 500 500
+
```
Comment (lukech) [inline test/evergreen.yml:2406]: Can we move this new block of code down, outside of the stress task definitions?

File: test/evergreen.yml [hunk 2]
```diff
@@ -2386,6 +2386,24 @@ tasks:
           times: 1       # No of times to run the loop
           no_of_procs: 10 # No of processes to run in the background
 
+  - name: performance-test
```
Comment (lukech) [inline test/evergreen.yml:2389]: Can we use a more specific task name, e.g. `many-handle-stress-test`?

File: test/evergreen.yml (hunk 1 above)
Comment (lukech) [reply test/evergreen.yml:2406]: I just realized it's a new perf task with stress as part of the script name. Anyway, I feel it's more of a 'perf' test and should be set in its own 'section' in the evergreen configuration file.

File: test/evergreen.yml [hunk 3]
```diff
@@ -2386,6 +2386,24 @@ tasks:
...
+    tags: ["performance-test-1"]
```
Comment (lukech) [inline test/evergreen.yml:2390]: Why do we need this tag? Tags are usually used to group tasks together, making calling them easier (by using tag selector feature).

File: test/evergreen.yml [hunk 4]
```diff
@@ -2659,6 +2677,23 @@ buildvariants:
     - name: ".stress-test-3"
     - name: ".stress-test-4"
 
+- name: performance-tests
+  display_name: Ubuntu 18.04 Performance tests
```
Comment (lukech) [inline test/evergreen.yml:2681]: This feels to be a very general variant name. I'm not sure if a more specific name e.g. `large scale tests` would make more sense here considering it only includes one perf correctness task right now. 
@sulabhM what's your opinion on this?

File: test/evergreen.yml [hunk 5]
```diff
@@ -2659,6 +2677,23 @@ buildvariants:
...
+  run_on:
+  - ubuntu1804-test
+  expansions:
+    smp_command: -j $(grep -c ^processor /proc/cpuinfo)
+    make_command: PATH=/opt/mongodbtoolchain/v3/bin:$PATH make
+    test_env_vars:
+      LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libeatmydata.so LD_LIBRARY_PATH=$(pwd)/.libs
+      PATH=/opt/mongodbtoolchain/v3/bin:$PATH top_srcdir=$(pwd)/.. top_builddir=$(pwd)
```
Comment (lukech) [inline test/evergreen.yml:2689]: The `test_env_vars` setting here seems not used by the task included in this build variant, and can be removed?

File: test/evergreen.yml [hunk 6]
```diff
@@ -2659,6 +2677,23 @@ buildvariants:
...
+    posix_configure_flags:
+      --enable-silent-rules --enable-python --enable-zlib --enable-snappy
+      --enable-strict --enable-static --prefix=$(pwd)/LOCAL_INSTALL
+    python_binary: '/opt/mongodbtoolchain/v3/bin/python3'
```
Comment (lukech) [inline test/evergreen.yml:2693]: Ditto

File: test/evergreen.yml [hunk 7]
```diff
@@ -2659,6 +2677,23 @@ buildvariants:
...
+  tasks:
+    - name: ".performance-test-1"
```
Comment (lukech) [inline test/evergreen.yml:2695]: As commented above, I don't think it worth using tag & tag selector for a single task inclusion. Listing the single new task here feels more intuitive and obvious.

File: test/evergreen.yml [hunk 8]
```diff
@@ -2659,6 +2677,23 @@ buildvariants:
     - name: ".stress-test-3"
     - name: ".stress-test-4"
 
+- name: performance-tests
+  display_name: Ubuntu 18.04 Performance tests
+  run_on:
+  - ubuntu1804-test
+  expansions:
+    smp_command: -j $(grep -c ^processor /proc/cpuinfo)
```
Comment (lukech) [inline test/evergreen.yml:2685]: Not sure it's used by the tasks in this build variant. Can we remove it?

File: test/evergreen.yml (hunk 2 above)
Comment (etienneptl) [reply test/evergreen.yml:2389]: Sure, I will rename it `many-dhandle-stress-test` to reflect the python script used.

File: test/evergreen.yml (hunk 3 above)
Comment (etienneptl) [reply test/evergreen.yml:2390]: I was thinking that if we keep creating stress testings, creating tags right now could prepare that.

File: test/evergreen.yml (hunk 4 above)
Comment (etienneptl) [reply test/evergreen.yml:2681]: We could effectively reused the terminology used for the million collection test (see #6681).
`name: large-scale-tests`
For the display name, we could also reuse `~Large scale tests`. What is the `~` though ?

**Review by lukech — APPROVED** [review]
LGTM

File: test/evergreen.yml (hunk 4 above)
Comment (lukech) [reply test/evergreen.yml]: Sounds good. The leading `~` is used for those least important or most expensive build variants, which does not seem to apply here.

File: test/evergreen.yml (hunk 3 above)
Comment (lukech) [reply test/evergreen.yml]: Got it. I think I prefer the more lazy approach - applying tags whenever there's a real need :)

File: test/evergreen.yml [hunk 9]
```diff
@@ -2659,6 +2676,18 @@ buildvariants:
     - name: ".stress-test-3"
     - name: ".stress-test-4"
 
+- name: large-scale-tests
```
Comment (sulabhM) [inline test/evergreen.yml:2679]: can we call it "performance tests", this will end up our perf tests on evergreen

**Review by sulabhM — APPROVED** [review]
LGTM with one comment

File: test/evergreen.yml (hunk 9 above)
Comment (etienneptl) [reply test/evergreen.yml:2679]: As discussed, will be renamed at a later stage if needed. Good for now.

**Pull request merged**